designs/embed_eksd_manifest.md

```diff
@@ -0,0 +1,90 @@
+# Embed EKS-D resources in EKS-A cluster
+
+## Introduction
+
+**Problem:** Currently, the EKS Anywhere controller pulls the EKS Distro manifest from the release URL provided in the versions bundle, which requires access to the internet.
+It would be ideal to have this manifest and its CRDs embedded in the cluster to allow the controller to operate without the requirement to make that external call.
+This will eventually be needed for air-gapped support.
+
+### Goals and Objectives
+
+As an EKS Anywhere user:
+
+* I want the EKSA controller to be able to operate without internet access
+* I want EKSD components to be supported during upgrade
+
+## Overview of Solution
+
+With this feature, the EKSD manifest & its CRD will be applied directly to the cluster.
+This will allow the controller to grab the EKSD release resources from within the cluster instead of reading in the EKSD manifest from the internet when fetching the applied spec.
+We will do this similarly to how we apply the EKSA components to the cluster, introducing a new task to the create workflow called `InstallEksdComponentsTask`.
+In the upgrade workflow, we will also add a step to apply the EKSD resources as we update other cluster resources.
+
+#### EKS-D bundle changes
+
+We currently don’t pull in the EKSD release CRD in the bundle, but we need to apply this CRD before applying the release manifest.
+I propose introducing a new field `components` in the EKSD bundle to refer to the file in S3 that stores this CRD.
+
+While most of our bundles have a `version` field, the EKS-D bundle does not have this field.
+Instead, the version is typically decided by the `releaseChannel` and the `releaseNumber`.
+We already have the `channel` field in the EKS-D bundle, so I propose adding the `number` field to this bundle to have an effective way to check the EKS-D version during upgrades.
+
+Example bundles manifest with proposed changes:
+
+```
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Bundles
+metadata:
+  creationTimestamp: "2022-02-01T16:41:19Z"
+spec:
+  cliMaxVersion: v0.0.0
+  cliMinVersion: v0.0.0
+  number: 1
+  versionsBundles:
+    eksD:
+      channel: 1-21
+      components: https://distro.eks.amazonaws.com/crds/releases.distro.eks.amazonaws.com-v1alpha1.yaml
+      gitCommit: |
+        fcddf59a516102e20f75b6c672e3aa74cf2da877
+      kindNode:
+        arch:
+          ...
+      kubeVersion: v1.21.5
+      manifestUrl: https://distro.eks.amazonaws.com/kubernetes-1-21/kubernetes-1-21-eks-8.yaml
+      name: kubernetes-1-21-eks-8
+      number: 8
+      ova:
+         ...
+```
+
+From here, we can use `kubectl apply` to apply the content from the EKSD release CRD & manifest to the cluster.
```

Comment:
what `manifest` are we applying?

Reply:
the eks-d manifest linked in the eks-d bundle, for example, `https://distro.eks.amazonaws.com/kubernetes-1-21/kubernetes-1-21-eks-8.yaml`